### ISSUE TITLE
Special Price From/To Times One Day not respected when sent/updated via Soap API

### ISSUE THREAD

**Issue by lano-vargas** [magento/magento2#19245]
Brief descritpion:

If I need the product to have special price for at least one day that ends at 3 o'clock so:
 ```
.....
<priceFrom>2018-11-23 00:00:00</priceFrom>
 <priceTo>2018-11-23 15:00:00</priceTo>
.....
```
### Preconditions (*)

1.Magento2.2.2
2.services=catalogSpecialPriceStorageV1

### Steps to reproduce (*)
1.Create configurable product and add at least one configurable option.
2.Add special price via API:

```
`<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:def="https://magento2/soap/all?services=catalogSpecialPriceStorageV1">
  <soap:Body>
    <def:catalogSpecialPriceStorageV1UpdateRequest>
      <prices>
        <item>
          <sku>configurable-option-sku</sku>
          <priceFrom>2018-11-23 00:00:00</priceFrom>
          <priceTo>2018-11-23 15:00:00</priceTo>
          <price>136.49</price>
          <storeId>0</storeId>
        </item>
      </prices>
    </def:catalogSpecialPriceStorageV1UpdateRequest>
  </soap:Body>
</soap:Envelope>`

```

### Expected result (*)
1. 
<img width="425" alt="screen shot 2018-11-16 at 11 39 39" src="https://user-images.githubusercontent.com/3495481/48619436-61338c80-e994-11e8-9b38-d53aa66cffc2.png">

2. The original price to be applied

### Actual result (*)

1.
<img width="458" alt="screen shot 2018-11-16 at 11 36 32" src="https://user-images.githubusercontent.com/3495481/48619348-1e71b480-e994-11e8-900b-579a5eb18d87.png">

2. The special price still applied

**HINT:** After all that fail go to backend **configurable-option-sku** product and just hit save then the original price is applied to the frontend

**Comment by magento-engcom-team** [comment]
Hi @lano-vargas. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@lano-vargas do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @lano-vargas thank you for you report this issue has already fixed in 2.3-develop branch.

**Comment by lano-vargas** [comment]
@engcom-backlog-nazar  Sorry Can you open and check again please now  as it about the times not being used !

**Comment by ghost** [comment]
@lano-vargas ok.

**Comment by ghost** [comment]
Hi @lano-vargas, thank you for you report, this issue has already fixed in 2.3-develop branch and will be available on 2.3.1 release.

**Comment by Ctucker9233** [comment]
@engcom-backlog-nazar This issue was reported against 2.2. Unless the fix is BiC, shouldn't this remain open until a fix is delivered to 2.2?

**Comment by ghost** [comment]
Hi @Ctucker9233 according to current workflow we close ticket if it was fixed on 2.3-dev but this one also fixed in 2.2-dev and available on 2.2.8 release. can you recheck ?

**Comment by ghost** [comment]
@lano-vargas we are closing this issue due inactivity. Feel free to reopen this one, if issue still exists.
